File: java/messages/src/main/java/com/nedap/retail/messages/epc/v2/difference_list/DifferenceListResponse.java
package com.nedap.retail.messages.epc.v2.difference_list;

import java.util.List;

import org.joda.time.DateTime;

import com.nedap.retail.messages.article.Article;

public class DifferenceListResponse {

    public List<String> gtins;

    @org.codehaus.jackson.annotate.JsonProperty("erp_stock")
    public List<Integer> erpStock;

    @org.codehaus.jackson.annotate.JsonProperty("rfid_stock")
    public List<Integer> rfidStock;

    public List<Article> articles;

    @org.codehaus.jackson.annotate.JsonProperty("erp_stock_time")
    public DateTime erpStockTime;

    @org.codehaus.jackson.annotate.JsonProperty("rfid_stock_time")
    public DateTime rfidStockTime;

    @org.codehaus.jackson.annotate.JsonProperty("erp_gtin_quantity")
    public Integer erpGtinQuantity;

    @org.codehaus.jackson.annotate.JsonProperty("rfid_gtin_quantity")
    public Integer rfidGtinQuantity;

    public DifferenceListResponse() {
    }

    public DifferenceListResponse(final List<String> gtins, final List<Integer> erpStock,
            final List<Integer> rfidStock, final List<Article> articles, final DateTime erpStockTime,
            final DateTime rfidStockTime, final Integer erpGtinQuantity, final Integer rfidGtinQuantity) {
        this.gtins = gtins;
        this.erpStock = erpStock;
        this.rfidStock = rfidStock;
        this.articles = articles;
        this.erpStockTime = erpStockTime;
        this.rfidStockTime = rfidStockTime;
        this.erpGtinQuantity = erpGtinQuantity;
        this.rfidGtinQuantity = rfidGtinQuantity;
    }
}

File: csharp/NedapRetailApiClient/Nedap.Retail.Api/Epc.V2/DifferenceListResponse.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace Nedap.Retail.Api.Epc.V2
{
    /// <summary>
    /// Difference between ERP system stock and RFID stock
    /// </summary>
    public class DifferenceListResponse : JsonPrintableObject
    {
        internal DifferenceListResponse()
        {

        }

        /// <summary>
        /// The GTINs in the difference list.
        /// </summary>
        [JsonProperty("gtins")]
        public List<string> Gtins { get; private set; }

        /// <summary>
        /// ERP stock as defined in the request. Values are signed.
        /// </summary>
        [JsonProperty("erp_stock")]
        public List<int> ErpStock { get; private set; }

        /// <summary>
        /// RFID stock as defined in the request. Values are unsigned.
        /// </summary>
        [JsonProperty("rfid_stock")]
        public List<int> RfidStock { get; private set; }

        /// <summary>
        /// Article data (only if include_articles is true).
        /// If no article data for a GTIN is known, a null object is included in the array for that GTIN.
        /// </summary>
        [JsonProperty("articles")]
        public List<Article.V2.Article> Articles { get; private set; }

        /// <summary>
        /// Time of ERP stock used to generate the difference list.
        /// </summary>
        [JsonProperty("erp_stock_time")]
        public DateTime ErpStockTime { get; private set; }

        /// <summary>
        /// Time of RFID stock used to generate the difference list.
        /// </summary>
        [JsonProperty("rfid_stock_time")]
        public DateTime RfidStockTime { get; private set; }

        /// <summary>
        /// Number of GTINs in ERP stock.
        /// </summary>
        [JsonProperty("erp_gtin_quantity")]
        public int ErpGtinQuantity { get; private set; }

        /// <summary>
        /// Number of GTINs in RFID stock.
        /// </summary>
        [JsonProperty("rfid_gtin_quantity")]
        public int RfidGtinQuantity { get; private set; }
    }
}